book/src/todo_app_1.md

```diff
@@ -0,0 +1,231 @@
+# Building a Simple To-Do App
+
+After we have learned so many concepts, it is finally time to put them into practice.
+We are going to build a To-Do app!
+
+For now, we would already be satisfied with a minimal version.
+An entry to input new tasks and a list view to display them will suffice.
+Something like this:
+
+<div style="text-align:center"><img src="img/todo_app_1.png" /></div>
+
+This mockup can be described by the following composite template.
+
+<span class="filename">Filename: listings/todo_app/1/window/window.ui</span>
+
+```xml
+{{#rustdoc_include ../listings/todo_app/1/window/window.ui}}
+```
+
+In order to use the composite template, we create a custom widget.
+The `parent` is `gtk::ApplicationWindow`, so we inherit from it.
+As usual, we have to list all [ancestors](https://docs.gtk.org/gtk4/class.ApplicationWindow.html#ancestors) and [interfaces](https://docs.gtk.org/gtk4/class.ApplicationWindow.html#implements) apart from `GObject` and `GInitiallyUnowned`.
+
+<span class="filename">Filename: listings/todo_app/1/window/mod.rs</span>
+
+```rust
+{{#rustdoc_include ../listings/todo_app/1/window/mod.rs:glib_wrapper}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+Then we initialize the composite template for `imp::Window`.
+We store references to the entry, the list view as well as the model.
+This will come in handy when we later add methods to our window.
+After that, we add the typical boilerplate for initializing composite templates.
+We only have to assure that the `class` attribute of the template in `window.ui` matches `NAME`.
+
+<span class="filename">Filename: listings/todo_app/1/window/imp.rs</span>
+
+```rust
+{{#rustdoc_include ../listings/todo_app/1/window/imp.rs:struct_and_subclass}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+`main.rs` also does not hold any surprises for us.
+
+<span class="filename">Filename: listings/todo_app/1/main.rs</span>
+
+```rust
+{{#rustdoc_include ../listings/todo_app/1/main.rs:main}}
+```
+
+So far so good.
+The main user interface is done, but the entry does not react to input yet.
+Also, where would the input go?
+We have not even set up the list view yet.
```

Comment:
This is a bit confusing, perhaps "We have not even set up the list model yet." rather than view?